torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -698,53 +720,93 @@ void LLVMCodeGenImpl::visit(const Min* v) {
-      case CompareSelectOperation::kEQ:
-        cmp_ = irb_.CreateFCmpOEQ(lhs, rhs);
-        break;
-      case CompareSelectOperation::kNE:
-        cmp_ = irb_.CreateFCmpONE(lhs, rhs);
-        break;
-      case CompareSelectOperation::kGT:
-        cmp_ = irb_.CreateFCmpOGT(lhs, rhs);
-        break;
-      case CompareSelectOperation::kGE:
-        cmp_ = irb_.CreateFCmpOGE(lhs, rhs);
-        break;
-      case CompareSelectOperation::kLT:
-        cmp_ = irb_.CreateFCmpOLT(lhs, rhs);
-        break;
-      case CompareSelectOperation::kLE:
-        cmp_ = irb_.CreateFCmpOLE(lhs, rhs);
-        break;
-      default:
-        // TODO: change to a proper error report
-        throw std::runtime_error("invalid operator type");
+  auto genUnbiased = [this, v]() -> llvm::Value* {
+    v->lhs()->accept(this);
+    auto lhs = this->value_;
+    v->rhs()->accept(this);
+    auto rhs = this->value_;
+    v->ret_val1()->accept(this);
+    auto retval1 = this->value_;
+    v->ret_val2()->accept(this);
+    auto retval2 = this->value_;
+
+    auto type_used = v->lhs()->dtype().scalar_type();
+
+    llvm::Value* cmp_;
+    CompareSelectOperation cmp_op_ = v->compare_select_op();
+
+    if (is_integral(type_used)) {
+      cmp_ = irb_.CreateICmp(
+          llvm_comparison_predicate(cmp_op_, type_used), lhs, rhs);
+    } else if (is_floating_point(type_used)) {
+      cmp_ = irb_.CreateFCmp(llvm_fp_comparison_predicate(cmp_op_), lhs, rhs);
+    } else {
+      throw std::runtime_error("invalid type for CompareSelect");
     }
-  } else {
-    throw std::runtime_error("invalid type for CompareSelect");
-  }
 
-  value_ = irb_.CreateSelect(cmp_, retval1, retval2);
-  return;
+    return irb_.CreateSelect(cmp_, retval1, retval2);
+  };
+
+  auto genBiased = [this, v]() -> llvm::Value* {
+    v->lhs()->accept(this);
+    auto lhs = this->value_;
+    v->rhs()->accept(this);
+    auto rhs = this->value_;
+
+    auto cmp_type = v->lhs()->dtype().scalar_type();
+    auto cmp_op = v->compare_select_op();
+    llvm::Value* cmp;
+
+    if (is_integral(cmp_type)) {
+      cmp = irb_.CreateICmp(
+          llvm_comparison_predicate(cmp_op, cmp_type), lhs, rhs);
+    } else if (is_floating_point(cmp_type)) {
+      cmp = irb_.CreateFCmp(llvm_fp_comparison_predicate(cmp_op), lhs, rhs);
+    } else {
+      throw std::runtime_error("invalid type for CompareSelect");
+    }
+
+    auto lanes = v->lhs()->dtype().lanes();
+    if (lanes > 1) {
+      auto maskType = llvm::Type::getIntNTy(getContext(), lanes);
+      auto zero = llvm::ConstantInt::get(maskType, 0);
+      auto mask = irb_.CreateBitOrPointerCast(cmp, maskType);
+      cmp = irb_.CreateICmpNE(mask, zero);
+    }
+
+    auto then_block = llvm::BasicBlock::Create(getContext(), "then", fn_);
+    auto else_block = llvm::BasicBlock::Create(getContext(), "else", fn_);
+    auto end_block = llvm::BasicBlock::Create(getContext(), "block", fn_);
+    constexpr int32_t total_weight = 100000;
```

Comment:
What's the significance of 100000 here, couldn't it be 10 or even 1?

Reply:
We talked about this offline, but just for posterity: this feature doesn't seem well documented.  I basically swiped this usage from the llvm repo itself, e.g. [this instrumentation in the ASAN implementation](https://github.com/llvm/llvm-project/blob/cfeecdf7b6df9cd89488948b440f8deeb458104c/llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp#L1677).